src/remember.ts

```diff
@@ -0,0 +1,55 @@
+import { Operator, StrictSink } from './types';
+
+export function remember<T>(numRemembered: number = 1): Operator<T, T> {
+  return source => {
+    const marker = {};
+    let sinks: Array<StrictSink<T> | undefined> = [];
+    let lasts: T[] = Array(numRemembered).fill(marker);
+    let curr = -1;
+    let started = false;
+
+    const mkTalkback = (sink: StrictSink<T>) => () => {
+      sinks[sinks.indexOf(sink)] = void 0;
+    };
+
+    return (_, sink) => {
+      sinks.push(sink);
+
+      if (!started) {
+        source(0, (t, d) => {
+          if (t === 0) {
+            started = true;
+            for (let i = 0; i < sinks.length; i++) {
+              const sink = sinks[i]!;
+              sink(0, mkTalkback(sink));
+            }
+          } else {
+            if (t === 1) {
+              curr = (curr + 1) % numRemembered;
+              lasts[curr] = d;
+            }
+
+            let hasDeleted = false;
+            for (let i = 0; i < sinks.length; i++) {
+              const sink = sinks[i];
+              if (sink) sink(t, d);
+              else hasDeleted = true;
+            }
+
+            if (hasDeleted) {
+              sinks = sinks.filter(x => x !== undefined);
+            }
+          }
+        });
+      } else {
+        sink(0, mkTalkback(sink));
+        for (let i = 0; i < lasts.length; i++) {
+          let idx = Math.abs((curr + 1 + i) % numRemembered);
```

Comment:
:+1: Cool idea to loop around with the modulo, this is much better than `arr.push()` and `arr.shift()`.